test/testenv/appframework_utils.go

```diff
@@ -132,3 +132,75 @@ func GetAppFileList(appList []string, version int) []string {
 	}
 	return appFileList
 }
+
+// GetBundleHash Get the status of bundle push on each indexer
+func GetBundleHash(deployment *Deployment, podName string) (string, error) {
+	stdin := "/opt/splunk/bin/splunk show cluster-bundle-status --verbose -auth admin:$(cat /mnt/splunk-secrets/password)"
+	command := []string{"/bin/sh"}
+	stdout, _, err := deployment.PodExecCommand(podName, command, stdin, false)
+	if err != nil {
+		logf.Log.Error(err, "Failed to execute command on pod", "pod", podName, "command", command, "stdin", stdin)
+		return "", err
+	}
+	var bundleHash string
+	for _, line := range strings.Split(string(stdout), "\n") {
+		if strings.Contains(line, "checksum") {
+			bundleHash = strings.Split(line, ":")[1]
+		}
+	}
+	return bundleHash, nil
+}
+
+// GetBundlePushStatus Get the status of bundle push on each indexer
+func GetBundlePushStatus(deployment *Deployment, podName string, ns string, previousBundleHash string) (map[string]string, error) {
+	stdin := "/opt/splunk/bin/splunk show cluster-bundle-status --verbose -auth admin:$(cat /mnt/splunk-secrets/password)"
+	command := []string{"/bin/sh"}
+	stdout, _, err := deployment.PodExecCommand(podName, command, stdin, false)
+	if err != nil {
+		logf.Log.Error(err, "Failed to execute command on pod", "pod", podName, "command", command, "stdin", stdin)
+		return nil, err
+	}
+
+	header := "-indexer-"
+	bundleStatus := make(map[string]string)
+
+	var indexerString string
+	var status string
+	var bundleHash string
+	indexerFound := false
+	for _, line := range strings.Split(string(stdout), "\n") {
+		// Check for empty lines to prevent an error in logic below
+		if len(line) == 0 {
+			continue
+		}
+		// Look for indexer name in output
+		if !indexerFound {
+			if strings.Contains(line, header) {
+				indexerString = strings.Split(line, "\t")[0]
+				indexerFound = true
+			}
+		} else {
+			// Match bundle hash when we have
+			if previousBundleHash != "" {
+				// Check if bundle hash is different from previous hash
+				if strings.Contains(line, "active_bundle") {
+					bundleHash = strings.Split(line, "=")[1]
+					// Check if new bundle was pushed by comparing hash
+					if bundleHash == previousBundleHash {
+						logf.Log.Info("Bundle hash not updated", "old Bundle hash", previousBundleHash, "new Bundle hash", bundleHash)
```

Comment:
Is this an error if the `bundleHash` is the same as the `previousBundleHash`?  Seems like we skip it and leave it out of the `bundleStatus[]`.